src/main/java/io/lettuce/core/cluster/models/tracking/TrackingInfoParser.java

```diff
@@ -0,0 +1,75 @@
+package io.lettuce.core.cluster.models.tracking;
+
+import io.lettuce.core.protocol.CommandKeyword;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Parser for Redis <a href="https://redis.io/docs/latest/commands/client-trackinginfo/">CLIENT TRACKINGINFO</a> command output.
+ *
+ * @author Tihomir Mateev
+ * @since 7.0
+ */
+public class TrackingInfoParser {
+
+    /**
+     * Utility constructor.
+     */
+    private TrackingInfoParser() {
+    }
+
+    /**
+     * Parse the output of the Redis CLIENT TRACKINGINFO command and convert it to a {@link TrackingInfo}
+     *
+     * @param trackinginfoOutput output of CLIENT TRACKINGINFO command
+     * @return an {@link TrackingInfo} instance
+     */
+    public static TrackingInfo parse(List<?> trackinginfoOutput) {
+
+        verifyStructure(trackinginfoOutput);
+
+        List<?> flags = (List<?>) trackinginfoOutput.get(1);
```

Comment:
Once you change the return type to Map<String, String>, you won't need all this hardcoded magic to access fields in a flat array that represents Map.

Reply:
Agreed, but as I mentioned it required adding a new `CommandOutput` implementation
I am not personally against that, but it increases the scope of the change drastically

Reply:
HELLO command has similar output, so perhaps we can follow a similar approach with `GenericMapOutput` or something similar?

Reply:
Unfortunately the `GenericMapOutput` does not handle nested arrays.
I just verified that there is a bug in the HELLO command that never exhibits itself, because we never read the modules field in the result of the HELLO command.